### ISSUE TITLE
MSTest v3 nullability on assertions should be more permissive

### ISSUE THREAD

**Issue by Evangelink** [microsoft/testfx#1451]
## Describe the bug

Most of the assertions APIs in MSTest v2 were supporting null values and would fail in case of null without forcing `Assert.IsNotNull` before. In MSTest v3, most of the APIs are expecting non-nullable entries.

## Steps To Reproduce

```csharp
StringAssert.StartsWith(nullableString, "something"); // potential null warning
```

**Comment by Evangelink** [comment]
Looking at the code this doesn't seem like a regression or bad design because the API would always fail in case of null values so it's better to warn the user that we don't expect null otherwise the API would fail.

**Comment by Evangelink** [comment]
Fixed by #1467
